crates/factor-sqlite/src/runtime_config.rs

```diff
@@ -0,0 +1,39 @@
+use std::{collections::HashMap, sync::Arc};
+
+use serde::Deserialize;
+use spin_factors::{anyhow, FactorRuntimeConfig};
+
+use crate::ConnectionPool;
+
+#[derive(Deserialize)]
+#[serde(transparent)]
+pub struct RuntimeConfig {
+    pub store_configs: HashMap<String, StoreConfig>,
+}
+
+impl FactorRuntimeConfig for RuntimeConfig {
+    const KEY: &'static str = "sqlite_database";
+}
+
+#[derive(Deserialize)]
+pub struct StoreConfig {
+    #[serde(rename = "type")]
+    pub type_: String,
+    #[serde(flatten)]
+    pub config: toml::Table,
+}
+
+/// Resolves some piece of runtime configuration to a connection pool
+pub trait RuntimeConfigResolver {
```

Comment:
Would it make sense for this crate to provide an impl that "does what spin does today"?

Reply:
I think so - though I think it might make sense to put it behind a feature flag since the implementation isn't trivial, and if a user of this crate doesn't want it we shouldn't hurt their compile times. 

I'd like to do that in a separate PR though since I think this will open up new questions, and I'd rather not have to worry about the base implementation becoming out of date.